app/models/conversation.rb

```diff
@@ -0,0 +1,13 @@
+class Conversation < ActiveRecord::Base
+ belongs_to :sender, :foreign_key => :sender_id, class_name: ‘User’
+ belongs_to :recipient, :foreign_key => :recipient_id, class_name: ‘User’
+
+has_many :messages, dependent: :destroy
+validates_uniqueness_of :sender_id, :scope => :recipient_id
+scope :involving, -> (user) do
+ where("conversations.sender_id =? OR conversations.recipient_id  =?", user.id, user.id)
+ end
+ scope :between, -> (sender_id, recipient_id) do
+ where("(conversations.sender_id = ? AND conversations.recipient_id =?) OR (conversations.sender_id = ? AND conversations.recipient_id =?)", sender_id,recipient_id, recipient_id, sender_id)
+  end
```

Comment:
`end` at 12, 2 is not aligned with `-> (sender_id, recipient_id) do` at 10, 17 or `scope :between, -> (sender_id, recipient_id) do` at 10, 1